rfcs/checkbox-component.md

```diff
@@ -1,5 +1,75 @@
 # Checkbox Component
 
+### Usage
+
+#### Basic usage
+
+```javascript
+import * as React from 'react';
+import {Checkbox} from 'baseui/checkbox';
+
+export default () => <Checkbox />;
+```
```

Comment:
I think for form controls the basic usage should hopefully still show passing a value in and change handlers

```js
<Checkbox 
  checked={props.isEnabled}
  onChange={evt => props.setEnabled(evt.target.checked)}
>
  Enabled
</Button>
```